### ISSUE TITLE
Track features for v0.1.2

### ISSUE THREAD

**Issue by flipchan** [uptest-sc/uptest#60]
Fixed storagevalue raw_type bug:
#59

**Comment by flipchan** [comment]
beta, the pre built migration search : https://github.com/uptest-sc/uptest/issues/64

**Comment by flipchan** [comment]
https://github.com/uptest-sc/uptest/issues/65

**Comment by flipchan** [comment]
https://github.com/uptest-sc/uptest/issues/69

**Comment by flipchan** [comment]
https://github.com/uptest-sc/uptest/pull/70

**Comment by flipchan** [comment]
fixed read_wasm binary found after the submit-runtime-upgrade script was publish 
https://github.com/uptest-sc/uptest/issues/71

**Comment by flipchan** [comment]
removed read_wasm_binary and replace it:
```rust
- use libuptest::metadata::read_wasm_binary;
+ use libuptest::metadata::read_wasm_binary_correct;
```

**Comment by flipchan** [comment]
https://github.com/uptest-sc/uptest/issues/73

**Comment by flipchan** [comment]
depricate unused chains/ dir

**Comment by flipchan** [comment]
pallet_storage_diff struct added, storage-changes in cli working and new

**Comment by flipchan** [comment]
cli upgraded:
```
 ./target/release/uptest  -c ws://127.0.0.1:9944
Uptest command line tool
running ws host: "ws://127.0.0.1:9944"
Connected to chain
Gathered current storage types
Waiting for runtime upgrade
Runtime upgrade in block: 0xb91bc8ed09a7a3e0accbaa92720412f382f4ae211218acece279a251bd67189c
Having a coffee break before next block...
Scanning the new metadata for changes
Runtime upgraded from version: 108 to new version: 109
Pallet name:  "TemplateModule"
    Storage item name:  "Something" 
    Storage item type:  StorageValue 
    Old storage type:  Primitive(U32)
    New storage type: Primitive(U64)
    
Pallet name:  "TemplateModule"
    Storage item name:  "Something6" 
    Storage item type:  StorageValue 
    Old storage type:  Primitive(U128)
    New storage type: Primitive(Bool)
    
```

**Comment by flipchan** [comment]
https://github.com/uptest-sc/uptest-sc.github.io/pull/13

**Comment by flipchan** [comment]
Uptest code structure, moved uptest-cli to different folder, added readme for uptest-cli: https://github.com/uptest-sc/uptest/pull/78

**Comment by flipchan** [comment]
we have a delivery called "macro X" after doing some research, we decided to make into a callable function to make it more portable

**Comment by flipchan** [comment]
subxt feature introduced, libuptest::types::H256 is not compatible with subxt::utils::H256

**Comment by flipchan** [comment]
input gen feature:
https://github.com/uptest-sc/uptest/issues/77

**Comment by flipchan** [comment]
https://github.com/uptest-sc/uptest/pull/83

**Comment by flipchan** [comment]
### Uptest release v0.1.2:  
uptest-cli is now renamed to uptest-cli and not just "uptest".
uptest-cli moved to its own subfolder with plans to seperate it to its own repo:
https://github.com/uptest-sc/uptest/pull/78

Fixed bug #59, uptest can now properly detect the storage item type:
https://github.com/uptest-sc/uptest/pull/59

Beta feature of migration search is live, feature allows you to automatically search for prewritten migrations in pallets:
https://github.com/uptest-sc/uptest/issues/64

Submit runtime upgrade wasm file:
https://github.com/uptest-sc/uptest/issues/65
In order to make debugging easier we have created a new repo that allows the user to submit runtime upgrades using a sudo
transaction:  https://github.com/uptest-sc/submit-runtime-upgrade

Uptest cli added feature chain-info:
https://github.com/uptest-sc/uptest/issues/69
https://github.com/uptest-sc/uptest/pull/70

Fixed bug that does not properly read the wasm file in libuptest:
```rust
- use libuptest::metadata::read_wasm_binary;
+ use libuptest::metadata::read_wasm_binary_correct;
```
Depricating the old function and added a new one:
https://github.com/uptest-sc/uptest/issues/71


depricated the chains/ directory in libuptest:
https://github.com/uptest-sc/uptest/issues/40


Uptest now supports compairing the old storage type(before upgrade) vs the new one(set after upgrade):
https://github.com/uptest-sc/uptest/issues/73

Patched storage-changes uptest-cli function

Public documentation updated:
https://uptest-sc.github.io/

All example code has been fixed:
https://github.com/uptest-sc/uptest/issues/61

+ Subxt feature flag has been added to make libuptest more compatible with subxt
+ Instead of delivering a macro that will wait for an event then execute the transaction, we delivered a function that is compatible with subxt, allowing everyone to schedule a subxt transaction. This was a more flexible solution:
https://github.com/uptest-sc/uptest/pull/82


Subxt H256 is now compatible with libuptest's H256:
https://github.com/uptest-sc/uptest/pull/79


As a promised milestone 2 delivery, we have enabled libuptest to support automatically generating u128, u64, u32, u8, f64, f32, bool and accountid32 addresses. Making it easier for developers to create temporary values, used for debugging.
https://github.com/uptest-sc/uptest/blob/main/examples/examples/test_input_gen.rs
https://github.com/uptest-sc/uptest/issues/77
https://github.com/uptest-sc/uptest/pull/83
https://github.com/uptest-sc/uptest/issues/8


subxthelper feature flag has now been added to all feature flags:
https://github.com/uptest-sc/uptest/pull/84

**Comment by flipchan** [comment]
released!
